### PR TITLE
DateMath: Add before/after forms

### PR DESCRIPTION
**Pull request by GuiltyDolphin**
Added forms like `3 days before today`, `4 weeks after 3rd Jan` etc...

------
IA Page:  http://duck.co/ia/view/date_math 
[Maintainer](http://docs.duckduckhack.com/maintaining/guidelines.html): @GuiltyDolphin

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: lib/DDG/Goodie/DateMath.pm [hunk 1]
```diff
@@ -124,10 +124,18 @@ sub get_result_action {
     return build_result($result, $formatted_input);
 }
 
+my $what_re = qr/what ((is|was|will) the )?/i;
+
+my $dort_re = qr/(?<dort>date|time|day)/i;
```
Comment (mintsoft) [inline lib/DDG/Goodie/DateMath.pm]: Can we rename that to `day_or_time`; `dort` just sounds weird.

Comment (GuiltyDolphin) [inline lib/DDG/Goodie/DateMath.pm]: Hehe, sure.

**Comment by cj01101** [comment]
Hi all - I don't have time to maintain this anymore. Is there someone else that can?

**Comment by GuiltyDolphin** [comment]
@cj01101 Sure! I do quite a bit with this Goodie so I'd be happy to take it on.

**Comment by GuiltyDolphin** [comment]
@mintsoft I'm going to put this on hold in lieu of #2810, I'm pretty sure this form will be supported out-of-the-box, we'll see.

**Comment by GuiltyDolphin** [comment]
@mintsoft I'm going to handle this differently to the Dates role - so there won't be any conflict.

I'm done with this (working on another change locally that I may well wish to rebase onto this); anything to add?

**Comment by mintsoft** [comment]
LGTM :+1:

**Pull request merged**